### PR TITLE
Added system architecture to `winget --info`.

### PR DESCRIPTION
**Pull request by jedieaston**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Are you working against an Issue?

-----

Resolves https://github.com/microsoft/winget-cli/issues/1925 (yes, I know it was my issue. just had to leave myself a reminder :D)

This PR adds the system architecture to `winget --info`, cutting out the first question we usually ask when troubleshooting.

image

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-cli/pull/1937)

### CODE REVIEW

File: src/AppInstallerCLIPackage/Shared/Strings/en-us/winget.resw [hunk 1]
```diff
@@ -1272,4 +1272,7 @@ Please specify one of them using the `--source` option to proceed.</value>
   <data name="InvalidArgumentWithoutQueryError" xml:space="preserve">
     <value>The arguments provided can only be used with a query.</value>
   </data>
+  <data name="Architecture" xml:space="preserve">
+    <value>Architecture</value>
```
Comment (JohnMcPMS) [inline src/AppInstallerCLIPackage/Shared/Strings/en-us/winget.resw:1276]:
```suggestion
    <value>System Architecture</value>
```

File: src/AppInstallerCLICore/Commands/RootCommand.cpp [hunk 2]
```diff
@@ -173,6 +173,8 @@ namespace AppInstaller::CLI
             info << std::endl <<
                 "Windows: "_liv << Runtime::GetOSVersion() << std::endl;
 
+            info << Resource::String::Architecture << ": "_liv << Utility::ToString(Utility::GetSystemArchitecture()) << std::endl;
```
Comment (yao-msft) [inline src/AppInstallerCLICore/Commands/RootCommand.cpp:176]: We haven't enforced the output rule yet, but I think the Utility::ToString(Utility::GetSystemArchitecture()) needs to be wrapped inside LocIndString

File: src/AppInstallerCLIPackage/Shared/Strings/en-us/winget.resw [hunk 3]
```diff
@@ -1272,4 +1272,7 @@ Please specify one of them using the `--source` option to proceed.</value>
   <data name="InvalidArgumentWithoutQueryError" xml:space="preserve">
     <value>The arguments provided can only be used with a query.</value>
   </data>
+  <data name="Architecture" xml:space="preserve">
```
Comment (yao-msft) [inline src/AppInstallerCLIPackage/Shared/Strings/en-us/winget.resw:1275]: nit: shall we match the variable name with the value so we don't misuse this in the future?

File: src/AppInstallerCLICore/Commands/RootCommand.cpp (hunk 2 above)
Comment (JohnMcPMS) [reply src/AppInstallerCLICore/Commands/RootCommand.cpp:176]: `Utility::ToString(Architecture)` could probably be changed to return a loc independent view.  If it turns out we need to localize it in the future, that can be its own problem.

**Comment by jedieaston** [comment]
Sorry for the delay, I made it localization independent when the command is called. I tried to do as @JohnMcPMS suggested and just change the type to LocIndView, but since LocIndView is also defined inside of that project, it seemed to create a circular dependency somehow (although since LocIndView wasn't relying on anything in AppInstallerArchitecture.h, I'm not sure why). Does anyone have any ideas on how to fix that?

**Comment by yao-msft** [comment]
> Sorry for the delay, I made it localization independent when the command is called. I tried to do as @JohnMcPMS suggested and just change the type to LocIndView, but since LocIndView is also defined inside of that project, it seemed to create a circular dependency somehow (although since LocIndView wasn't relying on anything in AppInstallerArchitecture.h, I'm not sure why). Does anyone have any ideas on how to fix that?

I just tried adding `#include "winget/LocIndependent.h"` to AppInstallerArchitectures.h and modified the declaration and definition of the method. I did not see circular dependency in rebuilding. If somehow this still won't work for you, I'm fine with merging with current code. Just let me know. Thanks.

**Comment by jedieaston** [comment]
It was the missing header :(

Thanks @yao-msft.

File: src/AppInstallerCommonCore/Architecture.cpp [hunk 4]
```diff
@@ -186,23 +186,23 @@ namespace AppInstaller::Utility
...
     }
 
-    std::string_view ToString(Architecture architecture)
+    LocIndView ToString(Architecture architecture)
     {
         switch (architecture)
         {
         case Architecture::Neutral:
-            return "Neutral"sv;
+            return LocIndView("Neutral"sv);
```
[the remark this answers is not in the document]
Comment (jedieaston) [reply src/AppInstallerCommonCore/Architecture.cpp:194]: I tried to use them but was missing the namespace. Oops, my header game is weak this week.

**Comment by denelon** [comment]
Does this work in the Sandbox? I submitted an [Issue](https://github.com/NickeManarin/ScreenToGif/issues/1075#issuecomment-1039749711) for "Screen To Gif". I wanted to make sure we're checking the host architecture, for this scenario or at least be aware of limitations.

**Review by yao-msft — APPROVED** [review]

**Comment by jedieaston** [comment]
> Does this work in the Sandbox? I submitted an [Issue](https://github.com/NickeManarin/ScreenToGif/issues/1075#issuecomment-1039749711) for "Screen To Gif". I wanted to make sure we're checking the host architecture, for this scenario or at least be aware of limitations.

Sorry I didn't get to this sooner, yes it works in the Sandbox.

**Pull request merged**